### PR TITLE
MAP-1493 Add validation to setting signed op cap

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/SignedOperationCapacityService.kt [hunk 1]
```diff
@@ -31,6 +35,10 @@ class SignedOperationCapacityService(
   fun saveSignedOperationalCapacity(request: SignedOperationCapacityValidRequest): SignOpCapResult {
     var newRecord = true
 
+    val maxCap = locationService.getResidentialLocations(request.prisonId).prisonSummary?.maxCapacity ?: throw PrisonNotFoundException(request.prisonId)
+    if (maxCap < request.signedOperationCapacity) {
+      throw CapacityException(request.prisonId, "Signed operational capacity cannot be more than the establishment's maximum capacity of $maxCap", ErrorCode.SignedOpCapCannotBeMoreThanMaXCap)
+    }
```
Comment (GurnankCheema) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/SignedOperationCapacityService.kt:41]: Can maxCap be equal to signedOperationCapacity ?

**Review by GurnankCheema — APPROVED** [review]
LGTM except for question

**Review by jimbali — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/SignedOperationCapacityService.kt (hunk 1 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/SignedOperationCapacityService.kt:41]: Yes e.g. 12 < 12 = false

**Pull request merged**